### PR TITLE
CSHARP-3699: Bump maxWireVersion for MongoDB 5.0

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
JIRA: https://jira.mongodb.org/browse/CSHARP-3699
EG: https://evergreen.mongodb.com/version/60e644df3627e06703f955c2

### CODE REVIEW

File: tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterTests.cs [hunk 1]
```diff
@@ -325,8 +325,8 @@ public void AcquireServerSession_should_call_serverSessionPool_AcquireSession()
         [Theory]
         [InlineData(0, 0, false)]
         [InlineData(0, 0, true)]
-        [InlineData(10, 12, false)]
-        [InlineData(10, 12, true)]
+        [InlineData(14, 15, false)]
+        [InlineData(14, 15, true)]
```
Comment (MikalaiMazurenka) [inline tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterTests.cs:329]: I didn't add the tests as requested ([comment](https://jira.mongodb.org/browse/CSHARP-3699?focusedCommentId=3862427&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3862427)) because this test already does what intended, so I just bumped the wire version here. Let me know if you think a new test is required.

Comment (JamesKovacs) [reply tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterTests.cs:329]: Works for me. My concern was to ensure that we test with incompatible wire versions, not that we write a new test specifically for this ticket.

File: tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerDescriptionTests.cs [hunk 2]
```diff
@@ -257,6 +257,8 @@ public void Equals_should_return_true_when_all_fields_are_equal()
         [InlineData(new[] { 6, 7 }, true)]
         [InlineData(new[] { 7, 7 }, true)]
         [InlineData(new[] { 7, 8 }, true)]
+        [InlineData(new[] { 10, 13 }, true)]
+        [InlineData(new[] { 14, 15 }, false)]
```
Comment (JamesKovacs) [inline tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerDescriptionTests.cs:261]: We should add some test cases that overlap but exceed the high end just as we do with the low end. I'd suggest:
```
[InlineData(new[] { 10, 14 }, true)]
[InlineData(new[] { 13, 15 }, true)]
```

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
One minor suggestion to add a few more test cases to verify edge case behaviour of the wire version check. Otherwise LGTM.

File: tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerDescriptionTests.cs (hunk 2 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerDescriptionTests.cs:263]: Good point!
Done

**Review by MikalaiMazurenka — COMMENTED** [review]
Ready for review!

**Review by JamesKovacs — APPROVED** [review]
LGTM
